spec.proto

```diff
@@ -0,0 +1,55 @@
+/**
+ * CloudEvent Protobuf Format
+ *
+ * - Required context attributes are explicity represented.
+ * - Optional and Extension context attributes are carried in a map structure.
+ * - Data may be represented as binary, text, or protobuf messages.
+ */
+
+syntax = "proto3";
+
+package io.cloudevents.v1;
+
+import "google/protobuf/any.proto";
+import "google/protobuf/timestamp.proto";
+
+option java_package = "io.cloudevents.v1.proto";
+
+message CloudEvent {
+
+  // -- CloudEvent Context Attributes
+
+  // Required Attributes
+  string id = 1;
+  string source = 2; // URI-reference
+  string spec_version = 3;
+  string type = 4;
+
+  // Optional & Extension Attributes
+  map<string, CloudEventAttribute> attribute = 5;
+
+  // -- CloudEvent Data (Bytes, Text, or Proto)
+  oneof data_oneof {
+    bytes binary_data = 6;
+    string text_data = 7;
+    google.protobuf.Any proto_data = 8;
+  }
+
+  /**
+   * The CloudEvent specification defines
+   * seven atrribute types...
+   */
+
+  message CloudEventAttribute {
```

Comment:
```suggestion
  // The value of a CloudEvent attribute, with a separate oneof case
  // for each type defined in the CloudEvent specification.
  message CloudEventAttributeValue {
```

Note that I've specified that it's the value, because this isn't the whole attribute - an attribute has both a name and a value, and this is *only* the value part.

An alternative design would be to include the name here and make the `attributes` field a repeated field instead.

Reply:
Ack